File: duplicate-assert/python-unittest/DetailAstImplTest_testClearChildCountCache.py
import unittest

class DetailAstImplTest(unittest.TestCase):

    def test_clear_child_count_cache(self):
        parent = DetailAstImpl()
        child = DetailAstImpl()
        parent.set_first_child(child)

        clear_child_count_cache_methods = [
            child.set_next_sibling,
            child.add_previous_sibling,
            child.add_next_sibling
        ]

        for method in clear_child_count_cache_methods:
            start_count = parent.get_child_count()
            method(None)
            intermediate_count = Whitebox.get_internal_state(parent, "childCount")
            finish_count = parent.get_child_count()
            self.assertEqual(start_count, finish_count, "Child count has changed")
            self.assertEqual(float('-inf'), intermediate_count, "Invalid child count")

        start_count = child.get_child_count()
        child.add_child(None)
        intermediate_count = Whitebox.get_internal_state(child, "childCount")
        finish_count = child.get_child_count()
        self.assertEqual(start_count, finish_count, "Child count has changed")
        self.assertEqual(float('-inf'), intermediate_count, "Invalid child count")

File: duplicate-assert/CSharpXUnit/CSharpXUnit.Tests/Tests/DetailAstImplTest_testClearChildCountCache.cs
using Xunit;

public class DetailAstImplTest : AbstractModuleTestSupport
{
    [Fact]
    public void TestClearChildCountCache()
    {
        var parent = new DetailAstImpl();
        var child = new DetailAstImpl();
        parent.SetFirstChild(child);

        var clearChildCountCacheMethods = new List<Action<DetailAstImpl>>
        {
            child.SetNextSibling,
            child.AddPreviousSibling,
            child.AddNextSibling
        };

        foreach (var method in clearChildCountCacheMethods)
        {
            var startCount = parent.GetChildCount();
            method(null);
            var intermediateCount = (int)Whitebox.GetInternalState(parent, "childCount");
            var finishCount = parent.GetChildCount();
            Assert.Equal(startCount, finishCount, "Child count has changed");
            Assert.Equal(int.MinValue, intermediateCount, "Invalid child count");
        }

        var startCountChild = child.GetChildCount();
        child.AddChild(null);
        var intermediateCountChild = (int)Whitebox.GetInternalState(child, "childCount");
        var finishCountChild = child.GetChildCount();
        Assert.Equal(startCountChild, finishCountChild, "Child count has changed");
        Assert.Equal(int.MinValue, intermediateCountChild, "Invalid child count");
    }
}
